Support HQL Subquery Insert Clause

```java
//JPQLQueryFactory query;
final QShift shs = new QShift("sh");
final JPQLQuery<Tuple> jpqlQuery = JPAExpressions.select(shs.dateTimeInterval, shs.employeePosition, shs.positionType, shs.positionIndex, shs.lunch, shs.outstaff,
                                            shs.availableForAssignment, shs.status, shs.edited, shs.ftePositionGroup, shs.plannedAbsence, Expressions.constant(newPcr.getId()))
        .from(shs).where(shs.positionCategoryRosterId.eq(oldPcr.getId()));
final QShift sh = QShift.shift;
final InsertClause<?> insertClause = query.insert(sh).columns(sh.dateTimeInterval, sh.employeePosition, sh.positionType, sh.positionIndex, sh.lunch, sh.outstaff,
                                                              sh.availableForAssignment, sh.status, sh.edited, sh.ftePositionGroup, sh.plannedAbsence, sh.positionCategoryRoster)
        .select(jpqlQuery);
return insertClause.execute();
```
ok. im make on JPA and remove querydsl-sql.
after clean rebuild
`
Caused by: org.hibernate.hql.internal.ast.QuerySyntaxException: expecting OPEN, found 'shift' near line 1, column 19 [insert into Shift shift (shift.dateTimeInterval, shift.employeePosition, shift.positionType, shift.positionIndex, shift.lunch, shift.outstaff, shift.availableForAssignment, shift.status, shift.edited, shift.ftePositionGroup, shift.plannedAbsence, shift.positionCategoryRoster)
select sh.dateTimeInterval, sh.employeePosition, sh.positionType, sh.positionIndex, sh.lunch, sh.outstaff, sh.availableForAssignment, sh.status, sh.edited, sh.ftePositionGroup, sh.plannedAbsence
from ru.abcconsulting.domain.entity.roster.Shift sh
where sh.positionCategoryRosterId = ?1]
`

```java
final QShift shs = QShift.shift;
final JPQLQuery<Tuple> jpqlQuery = JPAExpressions.select(shs.dateTimeInterval, shs.employeePosition, shs.positionType, shs.positionIndex, shs.lunch, shs.outstaff,
                                    shs.availableForAssignment, shs.status, shs.edited, shs.ftePositionGroup, shs.plannedAbsence, Expressions.constant(newPcr.getId()))
.from(shs).where(shs.positionCategoryRosterId.eq(oldPcr.getId()));
final QShift sh = new QShift("sh");
final InsertClause<?> insertClause = query.insert(sh).columns(sh.dateTimeInterval, sh.employeePosition, sh.positionType, sh.positionIndex, sh.lunch, sh.outstaff,
                                                      sh.availableForAssignment, sh.status, sh.edited, sh.ftePositionGroup, sh.plannedAbsence, sh.positionCategoryRoster)
.select(jpqlQuery);
return insertClause.execute();
```
`Caused by: org.hibernate.hql.internal.ast.QuerySyntaxException: expecting OPEN, found 'sh' near line 1, column 19 [insert into Shift sh (sh.dateTimeInterval, sh.emp`
after swapped

Expressions.constant(newPcr.getId()) not includet to select from my example. this generate hibernate exception
`Caused by: org.hibernate.QueryException: number of select types did not match those for insert [insert into Shift (dateTimeInterval, employeePosition, positionType, positionIndex, lunch, outstaff, availableForAssignment, status, edited, ftePositionGroup, plannedAbsence, positionCategoryRoster)
select shift.dateTimeInterval, shift.employeePosition, shift.positionType, shift.positionIndex, shift.lunch, shift.outstaff, shift.availableForAssignment, shift.status, shift.edited, shift.ftePositionGroup, shift.plannedAbsence
from ru.abcconsulting.domain.entity.roster.Shift shift
where shift.positionCategoryRosterId = ?1]`

--- a/querydsl-jpa/src/main/java/com/querydsl/jpa/JPQLSerializer.java
+++ b/querydsl-jpa/src/main/java/com/querydsl/jpa/JPQLSerializer.java
@@ -260,16 +260,30 @@ public class JPQLSerializer extends SerializerBase<JPQLSerializer> {
         }
     }
 
-    public void serializeForInsert(QueryMetadata md, List<Path<?>> columns, List<Object> values, SubQueryExpression<?> query, Map<Path<?>, Expression<?>> inserts) {
+    private static String relativePathString(Expression<?> root, Path<?> path) {
+        StringBuilder pathString = new StringBuilder(path.getMetadata().getName().length());
+        while (path.getMetadata().getParent() != null && !path.equals(root)) {
+            if (pathString.length() > 0) {
+                pathString.insert(0, '.');
+            }
+            pathString.insert(0, path.getMetadata().getName());
+            path = path.getMetadata().getParent();
+        }
+        return pathString.toString();
+    }
+
+    public void serializeForInsert(QueryMetadata md, Collection<Path<?>> columns, List<Object> values, SubQueryExpression<?> query, Map<Path<?>, Expression<?>> inserts) {
         append(INSERT);
-        handleJoinTarget(md.getJoins().get(0));
+        final JoinExpression root = md.getJoins().get(0);
+        append(getEntityName(root.getTarget().getType()));
         append(" (");
         boolean first = true;
         for (Path<?> path : columns) {
             if (!first) {
                 append(", ");
             }
-            handle(path);
+
+            append(relativePathString(root.getTarget(), path));
             first = false;
         }
         append(")\n");
@@ -285,9 +299,8 @@ public class JPQLSerializer extends SerializerBase<JPQLSerializer> {
                 handle(value);
                 first = false;
             }
-        }
-
-        if (inserts != null && inserts.entrySet().size() > 0) {
+            append(")");
+        } else if (inserts != null && inserts.entrySet().size() > 0) {
             first = true;
             for (Map.Entry<Path<?>, Expression<?>> entry : inserts.entrySet()) {
                 if (!first) {

--- a/querydsl-jpa/src/main/java/com/querydsl/jpa/hibernate/HibernateInsertClause.java
+++ b/querydsl-jpa/src/main/java/com/querydsl/jpa/hibernate/HibernateInsertClause.java
@@ -85,7 +85,7 @@ public class HibernateInsertClause implements
     @Override
     public long execute() {
         JPQLSerializer serializer = new JPQLSerializer(templates, null);
-        serializer.serializeForInsert(queryMixin.getMetadata(), columns, values, subQuery, inserts);
+        serializer.serializeForInsert(queryMixin.getMetadata(), inserts.isEmpty() ? columns : inserts.keySet(), values, subQuery, inserts);
         Map<Object, String> constants = serializer.getConstantToLabel();
 
         Query query = session.createQuery(serializer.toString());
@@ -121,7 +121,7 @@ public class HibernateInsertClause implements
     @Override
     public String toString() {
         JPQLSerializer serializer = new JPQLSerializer(templates, null);
-        serializer.serializeForInsert(queryMixin.getMetadata(), columns, values, subQuery, inserts);
+        serializer.serializeForInsert(queryMixin.getMetadata(), inserts.isEmpty() ? columns : inserts.keySet(), values, subQuery, inserts);
         return serializer.toString();
     }
 
@@ -158,8 +158,8 @@ public class HibernateInsertClause implements
 
     @Override
     public HibernateInsertClause values(Object... v)  {
-        // TODO Auto-generated method stub
-        return null;
+        this.values.addAll(Arrays.asList(v));
+        return this;
     }
 
 

--- a/querydsl-jpa/src/main/java/com/querydsl/jpa/impl/JPAInsertClause.java
+++ b/querydsl-jpa/src/main/java/com/querydsl/jpa/impl/JPAInsertClause.java
@@ -74,7 +74,7 @@ public class JPAInsertClause implements InsertClause<JPAInsertClause> {
     @Override
     public long execute() {
         JPQLSerializer serializer = new JPQLSerializer(templates, entityManager);
-        serializer.serializeForInsert(queryMixin.getMetadata(), columns, values, subQuery, inserts);
+        serializer.serializeForInsert(queryMixin.getMetadata(), inserts.isEmpty() ? columns : inserts.keySet(), values, subQuery, inserts);
         Map<Object,String> constants = serializer.getConstantToLabel();
 
         Query query = entityManager.createQuery(serializer.toString());
@@ -93,7 +93,7 @@ public class JPAInsertClause implements InsertClause<JPAInsertClause> {
     @Override
     public String toString() {
         JPQLSerializer serializer = new JPQLSerializer(templates, entityManager);
-        serializer.serializeForInsert(queryMixin.getMetadata(), columns, values, subQuery, inserts);
+        serializer.serializeForInsert(queryMixin.getMetadata(), inserts.isEmpty() ? columns : inserts.keySet(), values, subQuery, inserts);
         return serializer.toString();
     }
 

--- a/querydsl-jpa/src/test/java/com/querydsl/jpa/IntegrationBase.java
+++ b/querydsl-jpa/src/test/java/com/querydsl/jpa/IntegrationBase.java
@@ -147,6 +147,22 @@ public class IntegrationBase extends ParsingTest implements HibernateTest {
     }
 
     @Test
+    public void insert3() {
+        session.save(new Cat("Bob",10));
+
+        QCat cat = QCat.cat;
+        QCat bob = new QCat("Bob");
+
+        long amount = insert(cat)
+                .columns(cat.name, cat.alive)
+                .select(JPAExpressions.select(bob.name, bob.alive).from(bob))
+                .execute();
+        assertEquals(1, amount);
+
+        assertEquals(1L, query().from(cat).where(cat.name.eq("Bobby")).fetchCount());
+    }
+
+    @Test
     public void update() {
         session.save(new Cat("Bob",10));
         session.save(new Cat("Steve",11));

